libdnf-cli/output/transaction_table.hpp

```diff
@@ -0,0 +1,146 @@
+    sort_pkgs_list(pkgs);
+    for (auto & pkg : pkgs) {
+        struct libscols_line *ln = scols_table_new_line(tb, parent);
+        scols_line_set_data(ln, COL_NEVRA, pkg.get_full_nevra().c_str());
+        scols_line_set_data(ln, COL_REPO, pkg.get_repo_id().c_str());
+        scols_line_set_data(ln, COL_SIZE, libdnf::cli::utils::units::format_size(pkg.get_size()).c_str());
+    }
+}
+
+template <class Goal>
+bool print_transaction_table(Goal & goal) {
+    // TODO (nsella) split function into create/print if possible
+    //static struct libscols_table * create_transaction_table(bool with_status) {}
+    auto installs_pkgs = goal.list_rpm_installs();
+    auto reinstalls_pkgs = goal.list_rpm_reinstalls();
+    auto upgrades_pkgs = goal.list_rpm_upgrades();
+    auto downgrades_pkgs = goal.list_rpm_downgrades();
+    auto removes_pkgs = goal.list_rpm_removes();
+    auto obsoleted_pkgs = goal.list_rpm_obsoleted();
+
+    if (installs_pkgs.empty() && reinstalls_pkgs.empty() && upgrades_pkgs.empty() &&
+        downgrades_pkgs.empty() && removes_pkgs.empty() && obsoleted_pkgs.empty()) {
+        std::cout << "Nothing to do." << std::endl;
+        return false;
+    }
+
+    struct libscols_line *ln;
+
+    struct libscols_table *tb = scols_new_table();
+    scols_table_new_column(tb, "Package",    0.7, SCOLS_FL_TREE);
+    scols_table_new_column(tb, "Repository", 0.2, SCOLS_FL_TRUNC);
+    scols_table_new_column(tb, "Size",       0.1, SCOLS_FL_RIGHT);
+    scols_table_enable_maxout(tb, 1);
+    struct libscols_symbols *sb = scols_new_symbols();
+    scols_symbols_set_branch(sb, " ");
+    scols_symbols_set_right(sb, " ");
+    scols_symbols_set_vertical(sb, " ");
+    scols_table_set_symbols(tb, sb);
+
+    if (!installs_pkgs.empty()) {
+        ln = scols_table_new_line(tb, NULL);
+        scols_line_set_data(ln, COL_NEVRA, "Installing:");
+        add_line_into_transaction_table(tb, ln, installs_pkgs);
+    }
+
+    if (!reinstalls_pkgs.empty()) {
+        ln = scols_table_new_line(tb, NULL);
+        scols_line_set_data(ln, COL_NEVRA, "Reinstalling:");
+        add_line_into_transaction_table(tb, ln, reinstalls_pkgs);
+    }
+
+    if (!upgrades_pkgs.empty()) {
+        ln = scols_table_new_line(tb, NULL);
+        scols_line_set_data(ln, COL_NEVRA, "Upgrading:");
+        add_line_into_transaction_table(tb, ln, upgrades_pkgs);
+    }
+
+    if (!removes_pkgs.empty()) {
+        ln = scols_table_new_line(tb, NULL);
+        scols_line_set_data(ln, COL_NEVRA, "Removing:");
+        add_line_into_transaction_table(tb, ln, removes_pkgs);
+    }
+
+    if (!downgrades_pkgs.empty()) {
+        ln = scols_table_new_line(tb, NULL);
+        scols_line_set_data(ln, COL_NEVRA, "Downgrading:");
+        add_line_into_transaction_table(tb, ln, downgrades_pkgs);
+    }
+
+    scols_print_table(tb);
+    scols_unref_symbols(sb);
+    scols_unref_table(tb);
+
+    std::cout << "Transaction Summary:\n";
+    if (installs_pkgs.size()) {
```

Comment:
if (`installs_pkgs.size() > 0`) or better `if (!installs_pkgs.empty())`

Reply:
Changed and squashed, thanks